core/server/master/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -211,9 +212,23 @@ Command workerHeartbeat(long workerId, Map<String, Long> capacityBytesOnTiers,
       List<Metric> metrics);
 
   /**
-   * @return the block ids of lost blocks in Alluxio
-   */
-  Set<Long> getLostBlocks();
+   * @param blockId the block ID
+   * @return whether the block is considered lost in Alluxio
+   * */
+  boolean hasLostBlock(long blockId);
+
+  /**
+   * Returns an {@link Iterator} over the lost blocks.
+   * Note that the iterator should not be shared across threads.
+   *
+   * @return an Iterator
+   * */
+  Iterator<Long> getLostBlocksIterator();
+
+  /**
+   * @return the number of lost blocks in Alluxio
+   * */
+  int getLostBlocksCount();
```

Comment:
This seems to be test only.
Can you replace this with guava's Iterators.size() for the sake of reducing methods? Or mark it with @VisibleForTest?

Reply:
Actually I started with not having this method and using `Iterators.size()` to loop this iterator. But later I realized if we need the size we would have to do the same (or worse, loop the iterator to get the size in something else like a heartbeating `ReplicationChecker`), so I added this method.

I prefer not reducing this method and marking `@VisibleForTest` for now. And later if something else starts to rely on this method, this annotation can be removed.